smac/utils/validate.py

```diff
@@ -84,35 +81,35 @@ def __init__(self, scenario: Scenario, trajectory: list, output: str,
             num_run = np.random.randint(MAXINT)
             self.rng = np.random.RandomState(seed=num_run)
 
-        self.rh = RunHistory(average_cost)  # update this rh with validation-runs
-
-    def _save_results(self, rh:RunHistory, filename:str):
+    def _save_results(self, rh:RunHistory, output):
         """ Helper to save results to file """
-        if self.output == "":
+        if output == "":
             self.logger.info("No output specified, validated runhistory not saved.")
-        # Check if a folder or a file is specified as output
-        if not self.output.endswith('.json'):
-            old = self.output
-            self.output = os.path.join(self.output, filename)
-            self.logger.debug("Output is \"%s\", changing to \"%s\"!", old,
-                              self.output)
-        base = os.path.split(self.output)[0]
+            return rh
+        base = os.path.split(output)[0]
         if not base == "" and not os.path.exists(base):
             self.logger.debug("Folder (\"%s\") doesn't exist, creating.", base)
             os.makedirs(base)
-        self.logger.info("Saving validation-results in %s", self.output)
-        rh.save_json(self.output)
+        rh.save_json(output)
+        self.logger.info("Saving validation-results in %s", output)
         return rh
 
-    def validate(self, config_mode:Union[str, typing.List[Configuration]]='def',
+    def validate(self,
+                 config_mode:Union[str, typing.List[Configuration]]='def',
                  instance_mode:Union[str, typing.List[str]]='test',
                  repetitions:int=1, n_jobs:int=1, backend:str='threading',
-                 runhistory:RunHistory=None, tae:ExecuteTARun=None):
+                 runhistory:RunHistory=None, tae:ExecuteTARun=None,
+                 output:str="") -> RunHistory:
         """
         Validate configs on instances and save result in runhistory.
 
+        Sideeffect: if output is specified, saves runhistory to specified
+        output-directory.
+
         Parameters
         ----------
+        output: string
+            path to runhistory to be saved
```

Comment:
Is this the path to the output directory (as implied by the docstring for the argument above) or the path to the json file where the output should be written to?

Reply:
This question is not yet addressed.

Reply:
Clarified the docstrings. It's the filename. However, if the suffix is not '.json', it is interpreted as directory and an appropriate filename gets appended.